Integration

JoeJanickiChSh opened the pull request.

JoeJanickiChSh committed:
Angle limit for bumpers

JoeJanickiChSh committed:
Validated grab

JoeJanickiChSh committed:
Drive integration (totally arm safety code)

Commit:
Tried to get current from intake and limited arm rotate by 25%

Commit:
Actually got current and created a bang bang controller for arm rotate

Commit:
Changed variables to be in robot map

Commit:
Made dont hit ground better

JoeJanickiChSh committed:
New controls

JoeJanickiChSh committed:
Week 0

Commit:
made changes to dont hit the ground code

Liammurray19 committed:
added comments and other small things

Commit:
fixed some Enum stuff

Commit:
Added encoder reset to arm rotate

Commit:
Changed current limits for intake

JoeJanickiChSh committed:
Reset gyro on startup

Commit:
More intake intigration

TimB-87 committed:
Update Enums

Commit:
Testing changes

Commit:
Fixed zero velocity check

TimB-87 committed:
Attempt to make degrees more accurate

TimB-87 committed:
Make degrees more accurate (AKA oops)

Commit:
Auto stash before rebase of "origin/arm-safety"

TimB-87 committed:
Small fix

Commit:
gave up

Commit:
Made giving up less bad

JoeJanickiChSh committed:
Zero out camera position

TimB-87 committed:
Pid function to set points

TimB-87 committed:
Merge remote-tracking branch 'origin/arm-safety' into arm-safety

msoucy committed:
Merge branch 'main' into arm-safety

TimB-87 committed:
More integration

JoeJanickiChSh committed:
Make odometry and vision work 🥳

JoeJanickiChSh committed:
Preserve robot rotation when driving

JoeJanickiChSh committed:
Change naming

TimB-87 committed:
moremore integration

JoeJanickiChSh committed:
Merge branch 'arm-safety' into vision-integration

TimB-87 committed:
Small change for integration

msoucy committed:
Remove override that prevents auto from being cancelled

TimB-87 committed:
Small Integration

JoeJanickiChSh committed:
Alt shift o

msoucy commented on src/main/java/frc/robot/EnumLevel.java:
```diff
@@ -0,0 +1,32 @@
+    // 41 7/8 in.
+    HIGH_SCORE(19.6, 92),
+
+    FLOOR_PICKUP(0, 0),
```
Is this number accurate?

JoeJanickiChSh replied:
yes

TimB-87 replied:
No, no it is not, just put that in there as we have not measured yet.

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -57,25 +56,35 @@ public void robotInit() {
-        copilotController.rightBumper().whileTrue(intake.cubeRelease());
-        copilotController.x().onTrue(intake.coneGrab());
-        copilotController.leftBumper().onTrue(intake.coneRelease());
+        // copilotController.leftBumper().onTrue(intake.coneRelease());
```
Commented out code

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -57,25 +56,35 @@ public void robotInit() {
-        copilotController.leftBumper().onTrue(intake.coneRelease());
+        // copilotController.leftBumper().onTrue(intake.coneRelease());
+        copilotController.a().onTrue(intake.grabTwo());
+        // .onFalse(intake.safeStateCmd());
```
Commented out code

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -57,25 +56,35 @@ public void robotInit() {
+        // .onFalse(intake.safeStateCmd());
+        copilotController.b().onTrue(intake.coneToggle());
+        copilotController.x().whileTrue(intake.cubeRelease());
+        // copilotController.a().onTrue(intake.coneGrab());
```
Commented out code

bot190 commented on src/main/java/frc/robot/Vision.java:
```diff
@@ -64,16 +68,17 @@ public Pose2d update() {
-                odometry.resetPosition(driveMap.gyro().getRotation2d(),
-                        getModulePositions(), pose);
+                // odometry.resetPosition(driveMap.gyro().getRotation2d(),
+                // getModulePositions(), pose);
```
Delete commented code

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -57,25 +56,35 @@ public void robotInit() {
         // Led
-        copilotController.povUp().whileTrue(led.setYellow());
-        copilotController.povDown().whileTrue(led.setPurple());
+        // copilotController.povUp().whileTrue(led.setYellow());
```
Commented out code (but this is probably fine for now since we don't have LEDs)

msoucy commented on src/main/java/frc/robot/Vision.java:
```diff
@@ -64,16 +68,17 @@ public Pose2d update() {
                         .plus(cameraToRobot.inverse()).toPose2d();
 
-                driveMap.gyro().setAngle(pose.getRotation().getDegrees());
+                // driveMap.gyro().setAngle(pose.getRotation().getDegrees());
```
Commented out code

msoucy commented on src/main/java/frc/robot/Vision.java:
```diff
@@ -64,16 +68,17 @@ public Pose2d update() {
 
-                odometry.resetPosition(driveMap.gyro().getRotation2d(),
-                        getModulePositions(), pose);
+                // odometry.resetPosition(driveMap.gyro().getRotation2d(),
```
Commented out code

bot190 commented on src/main/java/frc/robot/maps/FrostBiteMap.java:
```diff
@@ -103,16 +106,16 @@ encoderRR, rearRightSteer, new CSSparkMax(1,
                 new Translation3d(
                         Units.inchesToMeters(
-                                2.44),
+                                0),
```
Do we need this info somewhere? Can we use it or remove it?

msoucy replied:
Formatting?

bot190 commented on src/main/java/frc/robot/maps/FrostBiteMap.java:
```diff
@@ -133,7 +136,16 @@ public IntakeData.Map getIntakeMap() {
+        intakeMotor.addValidator(() -> {
+            final double current = Arrays.stream(intakeMotor.getCurrentAmps()).reduce(Double::sum).orElse(0.0);
+            return currentFilter.calculate(current) < 20;
+        });
```
We're not using this anymore, lets get rid of it.

msoucy commented on src/main/java/frc/robot/maps/FrostBiteMap.java:
```diff
@@ -133,7 +136,16 @@ public IntakeData.Map getIntakeMap() {
+
+        final LinearFilter currentFilter = LinearFilter.singlePoleIIR(1, 0.01);
+        intakeMotor.addValidator(() -> {
+            final double current = Arrays.stream(intakeMotor.getCurrentAmps()).reduce(Double::sum).orElse(0.0);
```
Do we actually want the sum, or do we want the average?

JoeJanickiChSh replied:
An average is pretty much just a sum and we only have one motor. If we happen to add any more we could just multiply the current limit.

msoucy commented on src/main/java/frc/robot/maps/FrostBiteMap.java:
```diff
@@ -133,7 +136,16 @@ public IntakeData.Map getIntakeMap() {
+            return currentFilter.calculate(current) < 20;
+        });
 
+        // intakeMotor.validateCurrent(10, 0.5);
```
Commented out code

msoucy commented on src/main/java/frc/robot/subsystems/ArmRotate.java:
```diff
@@ -41,29 +48,72 @@ public ArmRotate(ArmRotateMap map) {
+    public CommandBase resetZero(DoubleSupplier speed) {
+        return cmd().onExecute(() -> {
+            data.setPoint = DESCEND_SPEED;
+            ;
```
`;`?

bot190 commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -13,14 +13,16 @@
+    public final double SPEED = 0.4;
     private final double RETRACT_SPEED = -0.1;
-    final double pivotHeight = 46.654;
+    final double PIVOT_HEIGHT = 46.654;
```
This should not be here, use the value from the map.

bot190 commented on src/main/java/frc/robot/subsystems/Arm.java:
```diff
@@ -31,47 +33,30 @@ public Arm(ArmMap extendMap) {
-            return length;
-        }
+    public boolean intakeBelowGround() {
+        return PIVOT_HEIGHT - INTAKE_DEPTH_LIMIT < Math.cos(Math.toRadians(armAngle)) * (data.distanceInches + 42.3);
```
EWW Magic numbers!

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -54,9 +55,13 @@ public Pose2d getPose() {
     private Field2d field = new Field2d();
 
     private final PIDController rotationPID;
+    private final PIDController driveRotationPID;
```
What's the difference between these two variables? Some comments would be helpful here.

msoucy commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -80,6 +124,6 @@ public void safeState() {
 
         // Stops movement of air
-        getData().solenoidSetPoint = Value.kOff;
+        // getData().solenoidSetPoint = Value.kOff;
```
Commented out code

bot190 replied:
Lets uncomment this since we aren't trying to use this to stop the intake spinning

bot190 commented on src/main/java/frc/robot/subsystems/ArmRotate.java:
```diff
@@ -41,29 +48,72 @@ public ArmRotate(ArmRotateMap map) {
+        }).runsUntil(() -> Math.abs(angle - data.degrees) < 2.0).onEnd(() -> {
+            data.setPoint = 0;
+        });
+    }
```
We can get rid of this since we have the PID control working

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -80,22 +85,33 @@ public CommandBase rotateToAngle(Rotation2d angle, DoubleSupplier translateX, Do
+    public CommandBase setSpeedCoef(double fac, double rotationfac) {
         return runOnce(() -> {
             speedCoef = fac;
+            ROTATIONSLOWCOEF = rotationfac;
```
This variable should not be capitalized.

bot190 commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -42,6 +60,32 @@ public CommandBase cubeGrab() {
         });
     }
 
+    public CommandBase grab() {
+        return cmd().onInitialize(
+                () -> {
+                    getData().motorSetPoint = GRAB_SPEED;
+                }).runsUntil(() -> getMap().motor.errored()).onEnd(() -> {
+                    getData().motorSetPoint = 0;
+                    getData().solenoidSetPoint = Value.kForward;
+                });
+    }
```
Remove this and rename `grabTwo` to `grab`.

JoeJanickiChSh committed:
Remove unused code

JoeJanickiChSh committed:
Remove more unused code

Commit:
Fixed pull comments

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -80,22 +87,33 @@ public CommandBase rotateToAngle(Rotation2d angle, DoubleSupplier translateX, Do
+
+        double rotationInput = deadband.applyAsDouble(rotation);
+
+        if (Math.abs(rotationInput) < 0.1) {
```
I'd love to have a comment here explaining what this does

TimB-87 replied:
Joe said he'll do it later

Lucimoot committed:
Intake Collision Check 

Added Intake Collision check for pistons in robot

JoeJanickiChSh committed:
Fix bug where robot randomly speeds off in one direction

JoeJanickiChSh committed:
Fix bug where robot randomly speeds off in one direction

TimB-87 committed:
PID Implementation for arm extension

TimB-87 committed:
PID Implementation for arm extension

Commit:
Added stow

Commit:
Added stow

Commit:
Added more to stow command

Commit:
Added more to stow command

TimB-87 committed:
Small Fixes

TimB-87 committed:
Merge remote-tracking branch 'origin/vision-integration' into vision-integration

Lucimoot committed:
Fixed Intake Collision check

It actually works now

Commit:
Added stow position that will never be used

Commit:
Merge remote-tracking branch 'origin/vision-integration' into vision-integration

TimB-87 committed:
Button Binding Fix

TimB-87 committed:
Merge remote-tracking branch 'origin/vision-integration' into vision-integration

Commit:
changed stow

TimB-87 committed:
Full Extend Preset

TimB-87 committed:
Merge remote-tracking branch 'origin/vision-integration' into vision-integration

TimB-87 committed:
Integration and Cleanup

TimB-87 committed:
Cleanup / Fix pull comments

JoeJanickiChSh committed:
A bunch of auto integration

JoeJanickiChSh committed:
Merge remote-tracking branch 'origin/vision-integration' into vision-integration

TimB-87 committed:
velocityPersistenceCheck Integration

TimB-87 committed:
Merge remote-tracking branch 'origin/vision-integration' into vision-integration

Commit:
Auto stash before merge of "vision-integration" and "origin/vision-integration"

JoeJanickiChSh committed:
DriveTowardCommand

JoeJanickiChSh committed:
Merge remote-tracking branch 'origin/vision-integration' into vision-integration

TimB-87 committed:
Angle reset while disabled

Commit:
Removed no fall for arm

TimB-87 committed:
Arm Rotate Integration

TimB-87 committed:
Clean up code (delete wrong stow)

Commit:
Auto stash before merge of "vision-integration" and "origin/vision-integration"

TimB-87 committed:
Fix Code

JoeJanickiChSh committed:
Distance limit

JoeJanickiChSh committed:
Merge remote-tracking branch 'origin/vision-integration' into vision-integration

TimB-87 committed:
Get and fix enum values

TimB-87 committed:
Set automatic scoring and change button bindings

(Made scoring for top and middle more automatic and changed a few button bindings to use the automatic scoring and fixed the resetZero() button binding.

JoeJanickiChSh committed:
Try to fix rotation correction

JoeJanickiChSh committed:
Merge remote-tracking branch 'origin/vision-integration' into vision-integration

TimB-87 committed:
Cleanup

msoucy commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -1,47 +1,85 @@
+    }
+
+    public CommandBase grab() {
+        return spinIn().andThen(new FunctionalWaitCommand(() -> 0.75), runOnce(() -> {
```
Why not just `WaitCommand` from wpilib? You're not using the double supplier for anything other than a constant

TimB-87 replied:
We need to use FunctionalWaitCommand because WaitCommand is broken and it crashes the code.

JoeJanickiChSh committed:
Deadband for translation

JoeJanickiChSh committed:
Merge remote-tracking branch 'origin/vision-integration' into vision-integration

JoeJanickiChSh committed:
Bring back camera offset
